### ISSUE TITLE
building scipy results in openblas not being found

### ISSUE THREAD

**Issue by krafczyk** [LLNL/spack#3007]
`spack install py-scipy^python@2.7.13` results in a build failure because numpy is not able to find libopenblas.

`spack-build.out`:

```
==> '/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/bin/python' 'setup.py' '--no-user-cfg' 'build'
Traceback (most recent call last):
  File "setup.py", line 415, in <module>
    setup_package()
  File "setup.py", line 395, in setup_package
    from numpy.distutils.core import setup
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/py-numpy-1.11.2-oedxdqtz445gq76ieasfss53ugpxe3ov/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/__init__.py", line 142, in <module>
    from . import add_newdocs
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/py-numpy-1.11.2-oedxdqtz445gq76ieasfss53ugpxe3ov/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/py-numpy-1.11.2-oedxdqtz445gq76ieasfss53ugpxe3ov/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/py-numpy-1.11.2-oedxdqtz445gq76ieasfss53ugpxe3ov/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/py-numpy-1.11.2-oedxdqtz445gq76ieasfss53ugpxe3ov/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/core/__init__.py", line 14, in <module>
    from . import multiarray
ImportError: libopenblas.so.0: cannot open shared object file: No such file or directory
```

**Comment by adamjstewart** [comment]
@davydden

**Comment by davydden** [comment]
i presume it's the current state of `develop` branch? Can you do activation/loading of `numpy` and check `np.show_config()`, see https://github.com/LLNL/spack/pull/2365 . 

I saw now issue with Ubuntu 16.04 + GCC5.4, what's the OS/compiler? Is `openblas` built by Spack?

**Comment by krafczyk** [comment]
After running `spack activate py-numpy; spack load python`, I check `np.show_config()` and get that error:

```
python -c 'import numpy as np; np.show_config()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/__init__.py", line 142, in <module>
    from . import add_newdocs
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/core/__init__.py", line 14, in <module>
    from . import multiarray
ImportError: libopenblas.so.0: cannot open shared object file: No such file or directory
```

OS = arch
compiler = 6.3.1
openblas is built by spack.

**Comment by davydden** [comment]
@krafczyk can you uninstall `numpy`, and re-install it but kill it in the middle and check `site.cfg`? If it still looks reasonable, I can only suggest filing an issue in numpy https://github.com/numpy/numpy/issues

**Comment by adamjstewart** [comment]
> re-install it but kill it in the middle

or run `spack install --keep-stage`

**Comment by krafczyk** [comment]
`site.cfg`:

```
[ALL]
rpath=/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/openblas-0.2.19-6x2e5bs2jyafzs7g47fu6nlga73wseo2/lib
[openblas]
libraries=openblas
library_dirs=/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/openblas-0.2.19-6x2e5bs2jyafzs7g47fu6nlga73wseo2/lib
```

@davydden I dont' believe this is numpy's problem however. If I do

`spack module loads --dependencies py-numpy`

then `python -c 'import numpy as np; np.show_config()'` works just fine.

```
python -c 'import numpy as np; np.show_config()'
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/openblas-0.2.19-6x2e5bs2jyafzs7g47fu6nlga73wseo2/lib']  
    define_macros = [('HAVE_CBLAS', None)]
    language = c
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/openblas-0.2.19-6x2e5bs2jyafzs7g47fu6nlga73wseo2/lib']  
    define_macros = [('HAVE_CBLAS', None)]
    language = c
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/openblas-0.2.19-6x2e5bs2jyafzs7g47fu6nlga73wseo2/lib']  
    define_macros = [('HAVE_CBLAS', None)]
    language = c
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/openblas-0.2.19-6x2e5bs2jyafzs7g47fu6nlga73wseo2/lib']  
    define_macros = [('HAVE_CBLAS', None)]
    language = c
blas_mkl_info:
  NOT AVAILABLE
```

In addition, if I manually add `openblas`'s library directory to my `LD_LIBRARY_PATH` after doing `spack activate py-numpy; spack load python`, then it works as well.

**Comment by davydden** [comment]
@krafczyk a whild guess, do you do `spack load python` before running everything? Maybe there is a mixture between system provided python and the one used to build `py-numpy`?

I was testing it by
```
$ spack activate py-numpy; spack load python;
$ python -c 'import numpy as np; t = np.ones((3,3),int); assert t.sum() == 9; assert np.dot(t,t).sum() == 27; np.show_config()'
```

**Comment by krafczyk** [comment]
Yes I do that. There is no overlap.

```
matthew on Wagner [15:28:21]
~/Software/NCSA/Vertical
○ spack activate py-numpy; spack load python;
==> Error: Package py-numpy-2cljb5d is already activated.

matthew on Wagner [15:28:31]
~/Software/NCSA/Vertical
○ which python
/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/bin/python

matthew on Wagner [15:28:38]
~/Software/NCSA/Vertical
○ python -c 'import numpy as np; t = np.ones((3,3),int); assert t.sum() == 9; assert np.dot(t,t).sum() == 27; np.show_config()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/__init__.py", line 142, in <module>
    from . import add_newdocs
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/home/matthew/Software/NCSA/Vertical/spack/spack/opt/spack/linux-arch-x86_64/gcc-6.3.1/python-2.7.13-k4lqbpgv5cfkmfrbcclaleymg54pb7nz/lib/python2.7/site-packages/numpy-1.11.2-py2.7-linux-x86_64.egg/numpy/core/__init__.py", line 14, in <module>
    from . import multiarray
ImportError: libopenblas.so.0: cannot open shared object file: No such file or directory

matthew on Wagner [15:28:52]
~/Software/NCSA/Vertical
○
```

**Comment by davydden** [comment]
@krafczyk hm... i would try `numpy` issue tracker for further help. `site.cfg` has `rpath` and `library_dirs` for blas, unfortunately at this stage i don't have in mind any further directions to try.

**Comment by adamjstewart** [comment]
`site.cfg` isn't something that needs to be copied to the installation directory is it? Is it only needed during build-time?

**Comment by davydden** [comment]
> Is it only needed during build-time?

that's my understanding of how things work.

**Comment by tgamblin** [comment]
Fixed by #3028
